akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/BehaviorSetup.scala

```diff
@@ -121,3 +121,59 @@ private[akka] object MDC {
     Map("persistenceId" -> persistenceId.id, "phase" -> phaseName)
   }
 }
+
+/**
+  * Setup snapshot and event delete/retention behavior. Retention bridges snapshot
+  * and journal behavior. This defines the retention criteria.
+  *
+  * @param snapshotEveryNEvents Snapshots are used to reduce playback/recovery times.
+  *                             This defines when a new snapshot is persisted.
+  *
+  * @param keepNSnapshots      After a snapshot is successfully completed,
+  *                             - if 2: retain last maximum 2 *`snapshot-size` events
+  *                             and 3 snapshots (2 old + latest snapshot)
+  *                             - if 0: all events with equal or lower sequence number
+  *                             will not be retained.
+  *
+  * @param deleteEventsOnSnapshot Opt-in ability to delete older events on successful
+  *                               save of snapshot. Defaults to disabled.
+  */
+/**
+  * Setup snapshot and event delete/retention behavior.
+  *
+  * INTERNAL API
+  */
+@InternalApi
+private[persistence]
+final case class RetentionCriteria private(snapshotEveryNEvents: Long,
+                                           keepNSnapshots: Long,
```

Comment:
`Int` is enough, we could use `Int.MaxValue` as a way to disable it completely

Reply:
I made these Long so that they can be in the future Retention trait for timestamp as TTL or KB size, for example. Why Int?

Reply:
Which is also why I originally had the name `snapshotSize` versus the later suggested `snapshotEveryNEvents`.

Reply:
It’s better to have different classes for the different algorithms than trying to squeze everything in same. That is what I mean with trait + class(es)

Reply:
That's not what I meant exactly, my thought was a trait with the three requirements. And for example 3 implementation case classes, each case class with the appropriate algorithm.